### PR TITLE
Enable TimelineEvent in GraalVM native image

### PR DESCRIPTION
**Pull request by jbachorik**
# What Does This Do
It enables TimelineEvent for native image

# Motivation
Providing code hotspots view for native image based apps

# Additional Notes

Jira ticket: [PROF-8742]

[PROF-8742]: https://datadoghq.atlassian.net/browse/PROF-8742?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/nativeimage/TracerActivation.java [hunk 1]
```diff
@@ -13,10 +15,20 @@ public final class TracerActivation {
...
+          TracerInstaller.installGlobalTracer(
+              new SharedCommunicationObjects(),
+              withProfiler
+                  ? new JFREventContextIntegration()
+                  : ProfilingContextIntegration.NoOp.INSTANCE);
+      // Need to start the profiling context integration explicitly
+      // When run as java agent this is done when the agent itself is started -
+      // for native image we need to run it ourselves
+      if (tracer != null) {
+        tracer.getProfilingContext().onStart();
```
Comment (richardstartin) [inline dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/nativeimage/TracerActivation.java:29]: I think we could probably refactor this (or elsewhere) to have a little less impact on our surroundings by initialising the context tracker differently depending on the context

Comment (jbachorik) [reply dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/nativeimage/TracerActivation.java:29]: Let me check what I can do without pulling off an extensive refactoring.

Comment (jbachorik) [reply dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/nativeimage/TracerActivation.java]: Ok, [this](https://github.com/DataDog/dd-trace-java/pull/6368/commits/24e618af4a101f1aba35195dbf1303266deeeac1) should simplify the activation a lot.

File: dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/TracerInstaller.java [hunk 2]
```diff
@@ -17,13 +17,14 @@ public static synchronized void installGlobalTracer(
...
-        installGlobalTracer(
+        CoreTracer tracer =
             CoreTracer.builder()
                 .sharedCommunicationObjects(sharedCommunicationObjects)
                 .profilingContextIntegration(profilingContextIntegration)
                 .pollForTracerFlareRequests()
                 .pollForTracingConfiguration()
-                .build());
+                .build();
+        installGlobalTracer(tracer);
```
Comment (richardstartin) [inline dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/TracerInstaller.java:27]: Is this change still necessary?

**Review by richardstartin — APPROVED** [review]

File: dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/TracerInstaller.java (hunk 2 above)
Comment (jbachorik) [reply dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/TracerInstaller.java:27]: Nope. Removed.
